### PR TITLE
XSAMPA conversion

### PR DESCRIPTION
**Pull request by annapiab**
Introducing a function that converts IPA transcriptions to X-SAMPA using Pynini. The conversion handles entire pronunciation strings, allowing spaces between phonemes with the `pynutil.join` function. The PR also includes corresponding test cases to ensure accuracy. The code is organized into two files: `xsampa.py` for the conversion logic and `xsampa_tests.py` for the unit tests.

### CODE REVIEW

File: xsampa.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import pynini
```
Comment (kylebgorman) [inline xsampa.py:1]: Put a module comment at the top indicating what this is. Something like:

```python
"""X-SAMPSA conversion functions for English.

Table based on: https://en.wikipedia.org/wiki/X-SAMPA
"""
```

File: xsampa.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+from pynini.lib import rewrite, pynutil
+
+# defining the IPA to X-SAMPA mappings as a list of tuples
```
Comment (kylebgorman) [inline xsampa.py:4]: This is a pretty self-evident comment, I'd cut it.

File: xsampa.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    ("ʊ", "U"), ("ʌ", "V"), ("ʍ", "W"), ("χ", "X"), ("ʏ", "Y"), ("ʒ", "Z"),
+    ("t͡s", "ts"), ("t͡ʃ", "tS"), ("t͡ɕ", "ts\\"), ("d͡ʒ", "dZ"),
+    ("l̩", "l="), ("n̩", "n="), ("ɝ", "<?"), ("ɪ̯", "I^"), ("ɫ̩", "5="),
+    ("aː", "a:"), ("eː", "e:"), ("iː", "i:"), ("oː", "o:"), ("uː", "u:"), 
+    ("æː", "{:"), ("ɑː", "A:"), ("ɔː", "O:"), ("ʊː", "U:"), ("ʌː", "V:"),
+    ("ɛː", "E:"), ("ɪː", "I:"), ("œː", "9:"), ("ɜː", "3:"), ("ʊ̯", "U^"), ("ɝː", "<? ɝ ?>:"),
+    ("m̩", "m_="), ("əː", "@:"), ("n̩", "n_=")
+]
+
+# pynini string map FST with space handling
```
Comment (kylebgorman) [inline xsampa.py:26]: This also strikes me as a self-evident comment; I'd cut it.

File: xsampa_test.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import unittest
+from xsampa import ipa_to_xsampa
```
Comment (kylebgorman) [inline xsampa_test.py:2]: Can you do `import xsampa` and then refer to the function as `ipa_to_xsampa`? That's consistent with what we do elsewhere in this library.

**Review by kylebgorman — COMMENTED** [review]
Looks good @annapiab. Can you take care of the few points I made here and then I'd like you to do the following:

1. Install `black` and `flake8` by running `pip install -r requirements.txt` in the top-level directory.
2. Run `black -l79` over both files.
3. Run `flake8` over both files and fix any lingering issues until it returns no issues.
4. Stage (i.e., `git add`), commit, and push your changes to this branch.

**Comment by kylebgorman** [comment]
Just thought of one other thing: this adds a Pynini dependency so we need to add that to `requirements.txt` and `pyproject.toml`. The version you want to use is `2.1.6.post1`.

**Comment by annapiab** [comment]
Hi Kyle, I made the chenges requested. Let me know if I forgot anything!

**Review by kylebgorman — COMMENTED** [review]
Looks good. I made a few very minor fixes:

1. The name of the package in the `requirementst.txt` and `pyproject.toml` should be lowercase "pynini" not "Pynini"
2. There was a shebang at the top of `xsampa.py`: it's harmless but that file is not actually executable so best to leave it off. 

I see one other issue: your `black` changes haven't gone through. Here's how I know:

$ black --check --line-length 79 xsampa.py xsampa_test.py
would reformat xsampa.py

Oh no! 💥 💔 💥
1 file would be reformatted, 1 file would be left unchanged.
$ flake8 xsampa.py xsampa_test.py
xsampa.py:14:68: W291 trailing whitespace
xsampa.py:15:78: W291 trailing whitespace
xsampa.py:24:74: W291 trailing whitespace
xsampa.py:26:80: E501 line too long (93 > 79 characters)
xsampa.py:30:80: E501 line too long (89 > 79 characters)
xsampa.py:32:1: E302 expected 2 blank lines, found 1

You need to run `black -l79 xsampa.py`, then run `flake8 xsampa.py` afterwards to ensure there are no issues left, then stage commit and push. Then you're done here.

**Comment by annapiab** [comment]
Hi Kyle, thanks for the patience. Let me know if this works!

**Review by kylebgorman — APPROVED** [review]
LGTM.

**Pull request merged**